selftest/test_images.py

```diff
@@ -7,129 +7,428 @@
-    image for image in os.getenv("TEST_CUSTOM_IMAGES", "").split(",") if image
+if not ARM64_VM_SIZES:
+    ARM64_VM_SIZES = [
+        sku.vm_size
+        for sku in selftest.SKU_CONFIGS.values()
+        if sku.vm_size_type == "arm64"
+    ]
+
+GEN1_IMAGES = [
+    image for image in os.getenv("SELFTEST_GEN1_IMAGES", "").split(",") if image
+]
+if not GEN1_IMAGES:
+    GEN1_IMAGES = [
+        "debian:debian-13-daily:13:latest",
+        "debian:debian-sid-daily:sid:latest",
+    ]
+GEN1_VM_SIZES = [
+    vm_size for vm_size in os.getenv("SELFTEST_GEN1_VM_SIZES", "").split(",") if vm_size
+]
+if not GEN1_VM_SIZES:
+    GEN1_VM_SIZES = [
+        sku.vm_size
+        for sku in selftest.SKU_CONFIGS.values()
+        if sku.vm_size_type == "x64" and not sku.vm_size.endswith("v6")
+    ]
+
+GEN2_IMAGES = [
+    image for image in os.getenv("SELFTEST_GEN2_IMAGES", "").split(",") if image
+]
+if not GEN2_IMAGES:
+    GEN2_IMAGES = [
+        "debian:debian-13-daily:13-arm64:latest",
+        "debian:debian-sid-daily:sid-arm64:latest",
+        "/CommunityGalleries/Fedora-5e266ba4-2250-406d-adad-5d73860d958f/Images/Fedora-Cloud-Rawhide-Arm64/versions/latest",
+    ]
+GEN2_VM_SIZES = [
+    vm_size for vm_size in os.getenv("SELFTEST_GEN2_VM_SIZES", "").split(",") if vm_size
 ]
-ENV_TEST_CUSTOM_VM_SIZES = os.getenv("TEST_CUSTOM_VM_SIZES", "")
-if ENV_TEST_CUSTOM_VM_SIZES == "DEFAULT_GEN1_VM_SIZES":
-    CUSTOM_VM_SIZES = DEFAULT_GEN1_VM_SIZES
-elif ENV_TEST_CUSTOM_VM_SIZES == "DEFAULT_GEN2_VM_SIZES":
-    CUSTOM_VM_SIZES = DEFAULT_GEN2_VM_SIZES
-elif ENV_TEST_CUSTOM_VM_SIZES == "DEFAULT_ARM64_VM_SIZES":
-    CUSTOM_VM_SIZES = DEFAULT_ARM64_VM_SIZES
-else:
-    CUSTOM_VM_SIZES = [image for image in ENV_TEST_CUSTOM_VM_SIZES.split(",") if image]
-
-
-def subprocess_run(cmd: List[str], check: bool = True):
+if not GEN2_VM_SIZES:
+    GEN2_VM_SIZES = [
+        sku.vm_size
+        for sku in selftest.SKU_CONFIGS.values()
+        if sku.vm_size_type == "x64"
+    ]
+
+
+def safe_parse_bytes_literal(s: str):
+    """Parse a bytes literal string (e.g. b'foo') safely to a string (foo)."""
+    s = s.strip()
+    if not s.startswith("b'") or not s.endswith("'"):
+        return s
+
+    # Strip the b' and trailing '
+    inner = s[2:-1]
+    # Decode escape sequences (e.g., \r, \n, \x00)
+    raw_bytes: bytes = codecs.escape_decode(inner.encode())[0]  # type: ignore
+    return raw_bytes.decode("utf-8", errors="ignore").rstrip("\x00")
+
+
+def _subprocess_run(
+    cmd: List[str],
+    *,
+    artifacts_path: Path,
+    artifact_name: str,
+    check: bool = False,
+    decode: bool = False,
+) -> subprocess.CompletedProcess:
     """Run a subprocess command and capture outputs as utf-8."""
-    print(f"executing command: {shlex.join(cmd)}")
-    try:
+    artifact_path = artifacts_path / artifact_name
+    deadline = datetime.now(timezone.utc) + timedelta(minutes=30)
+    while datetime.now(timezone.utc) < deadline:
+        printable_cmd = shlex.join(cmd)
```

Comment:
hoist this outside the while loop since the list of cmd is unchanged

Reply:
fixed